lib/processor.js

```diff
@@ -66,6 +67,136 @@ function getComment(html) {
     return html;
 }
 
+var leadingWhitespaceRegex = /^\s*/;
+
+/**
+ * When applying fixes, the postprocess step needs to know how to map fix ranges
+ * from their location in the linted JS to the original offset in the Markdown.
+ * Configuration comments and indentation trimming both complicate this process.
+ *
+ * Configuration comments appear in the linted JS but not in the Markdown code
+ * block. Fixes to configuration comments would cause undefined behavior and
+ * should be ignored during postprocessing. Fixes to actual code after
+ * configuration comments need to be mapped back to the code block after
+ * removing any offset due to configuration comments.
+ *
+ * Fenced code blocks can be indented by up to three spaces at the opening
+ * fence. Inside of a list, for example, this indent can be in addition to the
+ * indent already required for list item children. Leading whitespace inside
+ * indented code blocks is trimmed up to the level of the opening fence and does
+ * not appear in the linted code. Further, lines can have less leading
+ * whitespace than the opening fence, so not all lines are guaranteed to have
+ * the same column offset as the opening fence.
+ *
+ * The source code of a non-configuration-comment line in the linted JS is
+ * a suffix of the corresponding line in the Markdown code block. There are no
+ * differences within the line, so the mapping need only provide the offset
+ * delta at the beginning of each line.
+ *
+ * @param {string} text The text of the file.
+ * @param {ASTNode} node A Markdown code block AST node.
+ * @param {comments} comments List of configuration comment strings that will be
+ *     inserted at the beginning of the code block.
+ * @returns {number[]} A flattened list of 2-tuples, where the even-indexed
+ *     elements are range offsets in the linted JS used as a lookup key, and the
+ *     next odd-indexed element is the offset delta from linted JS to Markdown.
+ *     Add this value to a range from the linted JS to get the equivalent
+ *     location in the Markdown source.
+ */
```

Comment:
Just a small suggestion, do you think we can use a simple array of objects here like this `[{jsKey: 1,mdOffset:1},...]`?

I guess if we make it more explicit it will improve readability because took me a bit to get the concept here.

Reply:
Good idea. 9c221e7 implements this change. What do you think?

Reply:
Awesome, that's exactly what I had in mind 👍🏽